pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -1143,6 +1148,7 @@ def generateDrawSpecs(self, p):
                 #self.textHeight = height
                 offset = max(0,self.style['tickLength']) + textOffset
 
+                rect = QtCore.QRectF()
```

Comment:
are you instantiating here so mypy doesn't have a cow or is there some other reason?

If it's to make type-checking/editors happy, you can probably just do 

```python
rect: QtCore.QRectF  # no actual assignment
```

Reply:
This is only because I don't like having a variable instantiated within an `if` block, which doesn't have an `else` case.